### ISSUE TITLE
catch "connection closed" from camera-node possible? & text-data outside root-node

### ISSUE THREAD

**Issue by ozett** [henry-spanka/node-red-contrib-hikvision#19]
hi,
is it possible to filter/catch the "connection closed" message  in a flow?
i am asking, because i cannot see that the message is within msg.payload as an property (2).

how do i catch this within node-red?

![image](https://user-images.githubusercontent.com/5937811/70133587-b406e180-1686-11ea-91cd-32a04d083f03.png)

maybe there is room for improvement to send as a msg.property?

![image](https://user-images.githubusercontent.com/5937811/70133677-e0226280-1686-11ea-87c9-087818583cc9.png)

**Comment by henry-spanka** [comment]
You can check the message directly: if (msg == "Connection closed!") { ... }

**Comment by ozett** [comment]
👍 
will see if i can set up some sort of watchdog..
thx.

**Comment by ozett** [comment]
do you think that this will work within a node?

![image](https://user-images.githubusercontent.com/5937811/70604042-254f1300-1bf8-11ea-820d-93c0f5cbece9.png)

**Comment by ozett** [comment]
not really... 
![image](https://user-images.githubusercontent.com/5937811/70604139-60514680-1bf8-11ea-9d4d-fb9411a243b5.png)

can the "connection closed" get into a >>more<< node-red-stylish property for use in a switch-node?

**Comment by ozett** [comment]
i will try it with jsonata meanwhile...

![image](https://user-images.githubusercontent.com/5937811/70604709-5e3bb780-1bf9-11ea-85f6-25ffa5a63eb1.png)

**Comment by henry-spanka** [comment]
You need to set the output of the error node to "Complete message object" and not "msg.payload".

**Comment by ozett** [comment]
yeah, but most nodes want msg.payload, or i have to use change-node to move properties.
but that will work at the moment also, of course.
to have it in payload would make chaining some nodes a bit easyier. ..

**Comment by henry-spanka** [comment]
Try `msg.error`.

See: https://cookbook.nodered.org/basic/trigger-on-error#example

I don't think there is a way to set `msg.payload`. Which node requires `msg.payload`? Normally you can set the object namespace yourself.

**Comment by ozett** [comment]
i always make the mistake, that the inject-node injects only into msg.payload. 
a change-node has to follow to move it..

i try to avoid function-node and at firstplace use only the basic-nodes.
but that way i have to use too much change-nodes to transfer properties from/in outside of .payload.

but a good hint with the catch-node, as i believe, one could filter on catching only certain node..
i will try this, too.
